integration_tests/compose_api_test.go

```diff
@@ -0,0 +1,195 @@
+		}
+		brokerAPI = brokerapi.New(serviceBroker, logger, credentials)
+
+		responseRecorder = httptest.NewRecorder()
+		instanceID = uuid.NewV4().String()
+		bindingID = uuid.NewV4().String()
+		appGuid = uuid.NewV4().String()
+		paramJSON = "{}"
+		acceptsIncomplete = true
+	})
+
+	AfterEach(func() {
+		By("De-provisioning an instance", func() {
+			path := "/v2/service_instances/" + instanceID
+			request := newRequest(
+				"DELETE",
+				path,
+				nil,
+				uriParam{key: "service_id", value: serviceID},
+				uriParam{key: "plan_id", value: planID},
+				uriParam{key: "accepts_incomplete", value: strconv.FormatBool(acceptsIncomplete)},
+			)
+			brokerAPI.ServeHTTP(responseRecorder, request)
+			Expect(responseRecorder.Code).To(BeEquivalentTo(http.StatusAccepted))
+			body := readResponseBody(responseRecorder.Body)
+			var deprovisionResp brokerapi.DeprovisionResponse
+			err = json.Unmarshal(body, &deprovisionResp)
+			Expect(err).ToNot(HaveOccurred())
+			operationState := pollForOperationCompletion(instanceID, serviceID, planID, deprovisionResp.OperationData)
+			Expect(operationState).To(Equal("succeeded"), "returns success")
+		})
+	})
+
+	It("uses Compose API", func() {
+		By("provisioning an instance", func() {
+			path := "/v2/service_instances/" + instanceID
+			mongoVersion := "3.2.11"
+			provisionDetailsJson := []byte(fmt.Sprintf(`
+				{
+					"service_id": "%s",
+					"plan_id": "%s",
+					"organization_guid": "test-organization-id",
+					"space_guid": "space-id",
+					"parameters": {
+						"disable_ssl": true,
+						"wired_tiger": true,
+						"version": "%s"
+					}
+				}
+			`, serviceID, planID, mongoVersion))
+			param := uriParam{key: "accepts_incomplete", value: strconv.FormatBool(acceptsIncomplete)}
+			request := newRequest("PUT", path, bytes.NewBuffer(provisionDetailsJson), param)
+			brokerAPI.ServeHTTP(responseRecorder, request)
+			Expect(responseRecorder.Code).To(BeEquivalentTo(http.StatusAccepted))
+			body := readResponseBody(responseRecorder.Body)
+			var provisionResp brokerapi.ProvisioningResponse
+			err = json.Unmarshal(body, &provisionResp)
+			Expect(err).ToNot(HaveOccurred())
+			operationState := pollForOperationCompletion(instanceID, serviceID, planID, provisionResp.OperationData)
+			Expect(operationState).To(Equal("succeeded"), "and returns success")
+		})
+		By("binding an instance", func() {
+			path := fmt.Sprintf("/v2/service_instances/%s/service_bindings/%s", instanceID, bindingID)
+			bindingDetailsJson := []byte(fmt.Sprintf(`
+				{
+					"service_id": "%s",
+					"plan_id": "%s",
+					"bind_resource": {
+						"app_guid": "%s"
+					},
+					"parameters": "%s"
+				}`,
+				serviceID,
+				planID,
+				appGuid,
+				paramJSON,
+			))
+			req := newRequest(
+				"PUT",
+				path,
+				bytes.NewBuffer(bindingDetailsJson),
+			)
+
+			brokerAPI.ServeHTTP(responseRecorder, req)
+			Expect(responseRecorder.Code).To(Equal(http.StatusAccepted))
+
+			var data struct {
```

Comment:
Ditto a previous comment about using `brokerapi.Credentials`.

From this aspect, the duplication between the two sets of tests feels a bit strange.